specs/api_cron_flow.md

```diff
@@ -0,0 +1,130 @@
+# Changes and Automation for the API Update Flow
+
+## Problem Statement
+
+Some of the external APIs that we use may not allow us to process all
+of our data records in a single batch. If we send too many requests to
+the API in a short time, our access may be throttled or suspended. We
+need to be able to update the various external APIs in a reasonable
+fashion.
+
+## Background and Significance
+
+There are currently three APIs that we use to enhance the data sets
+from the SBA.
+
+The first is an API to get the Yelp rating where possible.
+
+The second is a Google Civics API to get the Congressional district
+based on the borrower address.
+
+The third is an attempt to obtain the latitude and longitude for the
+borrower addresses.
+
+Given the size of our database (even just for the SFDO loan set), if
+we try to update all the ratings in a single run, the API providers
+may throttle our access. The Google Civics API has a daily limit of
+25,000 calls and one of the potential goecode APIs has a 2,500 daily
+limit.
+
+The request is to be able to chunk the output into small batches and
+schedule a cron job to run repeatedly until all records have been
+attempted.
```

Comment:
This is important anyway because API terms and conditions generally state we can't store the result for over ___ days. We need a way to refresh the data automatically.

Reply:
I forgot about the limits on storage.